Sensus.Shared.iOS/Callbacks/UILocalNotifications/UILocalNotificationNotifier.cs

```diff
@@ -37,22 +39,6 @@ public override void IssueNotificationAsync(string message, string id, bool play
         {
             SensusContext.Current.MainThreadSynchronizer.ExecuteThreadSafe(() =>
             {
-                // cancel any existing notifications with the given id
-                lock (_notifications)
-                {
-                    foreach (UILocalNotification notification in _notifications.ToList())
-                    {
-                        string notificationId = notification.UserInfo.ValueForKey(new NSString(NOTIFICATION_ID_KEY)).ToString();
-                        if (notificationId == id)
-                        {
-                            CancelNotification(notification, NOTIFICATION_ID_KEY);
-
-                            // TODO:  How do we prevent this collection from growing without bound?
-                            _notifications.Remove(notification);
-                        }
-                    }
-                }
-
```

Comment:
That's actually one the code changes I'm most iffy about...

I chose to remove NOTIFICATION_ID_KEY everywhere for three reasons.

1. When I did a find all on NOTIFICATION_ID_KEY I only found one "setter" which was a constant string used to identify if notifications were survey callbacks. I moved all these checks over to the new NotificationType enum.
2. We seem to use it inconsistently. Above the "id" that is passed in is the callback id (if I remember correctly). If we need the reset code maybe we should actually do a check against the CallbackId?
3. The name doesn't match the pattern of any of the other Keys (e.g. SENSUS_CALLBACK_ID_KEY). This made me think maybe it was added later after the fact without a lot of thought for how the rest of the callback code works? That may have been an incorrect assumption.

Those three aren't definitive.  I didn't find clear errors. I just found lots and lots of things that looked wrong to me. I kind of decided that removing it and fixing any bugs that might come up would be better than keeping it in.